webapp/channels/src/components/suggestion/switch_channel_provider.tsx

```diff
@@ -53,237 +58,267 @@ import {isGuest} from 'mattermost-redux/utils/user_utils';
+    }
+    const showSlug = (isPartOfOnlyOneTeam || channel.type === Constants.DM_CHANNEL) && channel.type !== Constants.THREADS;
+
+    return (
+        <SuggestionContainer
+            ref={ref}
+            id={`switchChannel_${channel.name}`}
+            data-testid={channel.name}
+            role='listitem'
+            {...props}
+        >
+            {icon}
+            <div className='suggestion-list__ellipsis suggestion-list__flex'>
+                <span className='suggestion-list__main'>
+                    <span className={classNames({'suggestion-list__unread': item.unread && !channelIsArchived})}>{name}</span>
+                    {showSlug && description && <span className='ml-2 suggestion-list__desc'>{description}</span>}
+                </span>
+                {customStatus}
+                {sharedIcon}
+                {tag}
+                {badge}
+                {!isPartOfOnlyOneTeam && teamName}
+            </div>
+        </SuggestionContainer>
+    );
+});
+SwitchChannelSuggestion.displayName = 'SwitchChannelSuggestion';
+
+type OwnProps = SuggestionProps<WrappedChannel>;
+
+function mapStateToPropsForSwitchChannelSuggestion(state: GlobalState, ownProps: OwnProps) {
     const channel = ownProps.item && ownProps.item.channel;
     const channelId = channel ? channel.id : '';
     const draft = channelId ? getPostDraft(state, StoragePrefixes.DRAFT, channelId) : false;
-    const user = channel && getUser(state, channel.userId);
-    const userImageUrl = user && Utils.imageURLForUser(user.id, user.last_picture_update);
-    let dmChannelTeammate = channel && channel.type === Constants.DM_CHANNEL && getDirectTeammate(state, channel.id);
-    const userItem = getUserByUsername(state, channel.name);
-    const status = getStatusForUserId(state, channel.userId);
-    const collapsedThreads = isCollapsedThreadsEnabled(state);
-    const team = getTeam(state, channel.team_id);
-    const isPartOfOnlyOneTeam = getMyTeams(state).length === 1;
 
-    if (channel && !dmChannelTeammate) {
+    let dmChannelTeammate;
+    if (isRealChannel(channel) && channel.type === Constants.DM_CHANNEL) {
+        dmChannelTeammate = getDirectTeammate(state, channel.id);
+    } else if (isFakeDirectChannel(channel)) {
         dmChannelTeammate = getUser(state, channel.userId);
     }
 
+    let status;
+    if (dmChannelTeammate) {
+        status = getStatusForUserId(state, dmChannelTeammate.id);
+    }
+
+    const collapsedThreads = isCollapsedThreadsEnabled(state);
+
+    let team;
+    if (isRealChannel(channel)) {
+        team = getTeam(state, channel.team_id);
+    }
+
+    const isPartOfOnlyOneTeam = getMyTeams(state).length === 1;
+
     return {
         channelMember: getMyChannelMemberships(state)[channelId],
         hasDraft: draft && Boolean(draft.message.trim() || draft.fileInfos.length || draft.uploadsInProgress.length),
-        userImageUrl,
```

Comment:
This is just calculated in the component instead of being passed in from `mapStateToProps`